### ISSUE TITLE
about the given results of the model detection of upidet_release.pth

### ISSUE THREAD

**Issue by thelandord** [Eaphan/BiProDet#22]
I tested the pre-trained model under pytorch1.10.0, cuda11.3,python=3.8, and the result is different from the published result, is it because of the version of pytorch?
![image](https://github.com/Eaphan/UPIDet/assets/103831660/cfd0fe43-ae3e-4045-9b61-a7835e9f13db)

![car](https://github.com/Eaphan/UPIDet/assets/103831660/59471857-ff6b-432a-8153-26935979e55d)
![cyc](https://github.com/Eaphan/UPIDet/assets/103831660/8b02e425-f82c-4f65-9f32-cc55e7de72f6)
![ped](https://github.com/Eaphan/UPIDet/assets/103831660/6ec0af5d-c184-46b2-bd8a-4125541b8139)

**Comment by Eaphan** [comment]
The reason for this discrepancy could be due to differences in library versions and the presence of noise introduced during pre-processing.

**Comment by thelandord** [comment]
> The reason for this discrepancy could be due to differences in library versions and the presence of noise introduced during pre-processing.

Thank you for your help!
I have another problem, I encountered the following bug while training, at line 161 of dataset.py
**data_dict['segmentation'] = [data_dict['segmentation'][ind] for ind in selected]**
There is a bug: IndexError: list index out of range.
 I print it and get the output:
![wenti](https://github.com/Eaphan/UPIDet/assets/103831660/8fd56d3a-6831-46f5-994d-e43ce8e54017)
Why is the len of data_dict['segmentation'] less than the len of selected? Do I need to align the selected filter to data_dict['segmentation']?

**Comment by Eaphan** [comment]
Please follow the README.md to prepare data and train.

**Comment by thelandord** [comment]
> Please follow the README.md to prepare data and train.

thanks for your reply！ I have replaced the .pkl file.I meet the bug after replaced pkl. But the len of data_dict['segmentation'] less than the len of selected.Is there a detail I'm missing?

**Comment by Eaphan** [comment]
Have you replaced all original pkl files, including the kitti_dbinfos_train.pkl.

**Comment by thelandord** [comment]
> Have you replaced all original pkl files, including the kitti_dbinfos_train.pkl.

yes, i have replaced the kitti_dbinfos_train.pkl, kitti_infos_train.pkl ,kitti_infos_val.pkl. i print the pkl as the following pictures 
![Snipaste_2024-06-26_21-27-09](https://github.com/Eaphan/UPIDet/assets/103831660/828dede8-b34d-4ec5-8920-a4168decfd6f)

**Comment by Eaphan** [comment]
Could you use the command "git diff" to see what changes you have made?

**Comment by Eaphan** [comment]
The issue could be due to the PyTorch version or jitter caused by random sampling of input point clouds.
